src/validation.cpp

```diff
@@ -4303,7 +4304,15 @@ bool LoadMempool(void)
                 if (state.IsValid()) {
                     ++count;
                 } else {
-                    ++failed;
+                    // mempool may contain the transaction already, e.g. from
+                    // wallet(s) having loaded it while we were processing
+                    // mempool transactions; consider these as valid, instead of
+                    // failed, but mark them as 'already existed'
+                    if (mempool.exists(tx->GetHash())) {
```

Comment:
Doesn't the above `LOCK(cs_main)` prevents that case (and the current implementation for all that matters)?